src/pages/incomplete.js

```diff
@@ -46,6 +52,22 @@ const Incomplete = () => {
       setError(true)
     }
   }
+
+  useEffect(() => {
+    if (isAuthenticated && !dataIsLoading && !isIncomplete) {
```

Comment:
I think it might be good to wait until data is loaded, as incomplete might be false also when data is loading.